### PR TITLE
Extracted azure features

### PR DESCRIPTION
**Pull request by lopushen**
Extracted the most commonly used azure bundles to features

### CODE REVIEW

File: features/src/main/feature/camel-features.xml [hunk 1]
```diff
@@ -168,7 +168,22 @@
         <bundle dependency='true'>wrap:mvn:software.amazon.awssdk/regions/${aws-java-sdk2-version}</bundle>
         <bundle dependency='true'>wrap:mvn:software.amazon.awssdk/utils/${aws-java-sdk2-version}</bundle>
     </feature>
-
+    <feature name="azure" start-level="50">
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
+        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
+        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
```
Comment (lopushen) [inline features/src/main/feature/camel-features.xml:175]: dependency tree suggests that azure core depends on the 2 bundles above - added and using the feature event when those bundles are not directly required by OSGI

File: features/src/main/feature/camel-features.xml [hunk 2]
```diff
@@ -490,92 +505,61 @@
...
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-messaging-eventhubs/5.18.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-messaging-eventhubs-checkpointstore-blob/1.19.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob/12.25.2</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.2</bundle>
-        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
-        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
+        <feature>azure-eventhubs</feature>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob/12.25.1</bundle>
```
Comment (lopushen) [inline features/src/main/feature/camel-features.xml]: There are 2 usages of `wrap:mvn:com.azure/azure-storage-blob/12.25.1` bundle, 2 as a single bundle, one along with `<bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob-changefeed/${azure-storage-blob-changefeed-version}</bundle>`
Can extract it to a separate feature as well - not sure if it is feasible

File: features/src/main/feature/camel-features.xml [hunk 3]
```diff
@@ -168,7 +168,22 @@
         <bundle dependency='true'>wrap:mvn:software.amazon.awssdk/regions/${aws-java-sdk2-version}</bundle>
         <bundle dependency='true'>wrap:mvn:software.amazon.awssdk/utils/${aws-java-sdk2-version}</bundle>
     </feature>
-
+    <feature name="azure" start-level="50">
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
    <feature name="azure" version="1.46.0" start-level="50">
```

File: features/src/main/feature/camel-features.xml [hunk 4]
```diff
@@ -168,7 +168,22 @@
...
         <bundle dependency='true'>wrap:mvn:software.amazon.awssdk/utils/${aws-java-sdk2-version}</bundle>
     </feature>
-
+    <feature name="azure" version="1.46.0" start-level="50">
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
+        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
+        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
+    </feature>
+    <feature name="azure-storage" start-level="50">
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
    <feature name="azure-storage" version="1.46.0" start-level="50">
```

File: features/src/main/feature/camel-features.xml [hunk 5]
```diff
@@ -168,7 +168,22 @@
...
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
+        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
+        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
+    </feature>
+    <feature name="azure-storage" version="1.46.0" start-level="50">
+        <feature>azure</feature>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.1</bundle>
+    </feature>
+    <feature name="azure-eventhubs" start-level="50">
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
    <feature name="azure-eventhubs" version="1.46.0" start-level="50">
```

File: features/src/main/feature/camel-features.xml [hunk 6]
```diff
@@ -168,7 +168,22 @@
...
     </feature>
-
+    <feature name="azure" version="1.46.0" start-level="50">
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
+        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
+        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
+    </feature>
+    <feature name="azure-storage" version="1.46.0" start-level="50">
+        <feature>azure</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 7]
```diff
@@ -168,7 +168,22 @@
...
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
+        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
+        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
+    </feature>
+    <feature name="azure-storage" version="1.46.0" start-level="50">
+        <feature version='[1.46,1.47)'>azure</feature>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.1</bundle>
+    </feature>
+    <feature name="azure-eventhubs" version="1.46.0" start-level="50">
+        <feature>azure-storage</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure-storage</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 8]
```diff
@@ -490,92 +505,61 @@
     <feature name='camel-azure-cosmosdb' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-cloudevents</feature>
         <feature version='[2.16,2.17)'>jackson</feature>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
+        <feature>azure</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 9]
```diff
@@ -490,92 +505,61 @@
...
         <bundle>mvn:org.apache.camel.karaf/camel-azure-cosmosdb/${project.version}</bundle>
     </feature>
     <feature name='camel-azure-eventhubs' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-cloudevents</feature>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core-amqp/2.9.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-messaging-eventhubs/5.18.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-messaging-eventhubs-checkpointstore-blob/1.19.1</bundle>
+        <feature>azure-eventhubs</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure-eventhubs</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 10]
```diff
@@ -490,92 +505,61 @@
...
-        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-azure-eventhubs/${project.version}</bundle>
     </feature>  
     <feature name='camel-azure-files' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-ftp</feature>
         <feature version='${camel.osgi.version.range}'>camel-cloudevents</feature>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.2</bundle>
+        <feature>azure-storage</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure-storage</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 11]
```diff
@@ -490,92 +505,61 @@
...
     </feature>
     <feature name='camel-azure-key-vault' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-core</feature>
         <feature version='[2.16,2.17)'>jackson</feature>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core-amqp/2.9.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-messaging-eventhubs/5.18.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-messaging-eventhubs-checkpointstore-blob/1.19.1</bundle>
+        <feature>azure-eventhubs</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure-eventhubs</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 12]
```diff
@@ -490,92 +505,61 @@
...
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.2</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-azure-key-vault/${project.version}</bundle>
     </feature>
     <feature name='camel-azure-schema-registry' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-core</feature>
-        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
-        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
+        <feature>azure</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 13]
```diff
@@ -490,92 +505,61 @@
...
         <bundle>mvn:org.apache.camel.karaf/camel-azure-servicebus/${project.version}</bundle>
     </feature>
     <feature name='camel-azure-storage-blob' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-cloudevents</feature>
-        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
-        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.45.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob/12.25.1</bundle>
+        <feature>azure-storage</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure-storage</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 14]
```diff
@@ -490,92 +505,61 @@
...
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.46.0</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.2</bundle>
+        <feature version='[1.46,1.47)'>azure</feature>
         <bundle>mvn:org.apache.camel.karaf/camel-azure-schema-registry/${project.version}</bundle>
     </feature>
     <feature name='camel-azure-servicebus' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-cloudevents</feature>
-        <bundle dependency='true'>mvn:org.reactivestreams/reactive-streams/${reactive-streams-version}</bundle>
-        <bundle dependency='true'>mvn:io.projectreactor/reactor-core/3.4.34</bundle>
+        <feature>azure</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 15]
```diff
@@ -490,92 +505,61 @@
...
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob/12.25.1</bundle>
+        <feature version='[1.46,1.47)'>azure-storage</feature>
+        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob/12.25.2</bundle>
         <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-blob-changefeed/${azure-storage-blob-changefeed-version}</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.1</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-azure-storage-blob/${project.version}</bundle>
     </feature>
     <feature name='camel-azure-storage-datalake' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-core</feature>
+        <feature>azure-storage</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure-storage</feature>
```

File: features/src/main/feature/camel-features.xml [hunk 16]
```diff
@@ -490,92 +505,61 @@
...
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.1</bundle>
         <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-file-datalake/12.18.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-identity/1.11.1</bundle>
         <bundle>mvn:org.apache.camel.karaf/camel-azure-storage-datalake/${project.version}</bundle>
     </feature>
     <feature name='camel-azure-storage-queue' version='${project.version}' start-level='50'>
         <feature version='${camel.osgi.version.range}'>camel-cloudevents</feature>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-core/1.45.1</bundle>
-        <bundle dependency='true'>wrap:mvn:com.azure/azure-storage-common/12.24.1</bundle>
+        <feature>azure-storage</feature>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]:
```suggestion
        <feature version='[1.46,1.47)'>azure-storage</feature>
```

**Review by essobedo — APPROVED** [review]
LGTM

**Pull request merged**